### ISSUE TITLE
`_undersample_data` method ignores `experiment_num` parameter, causing identical sampling results across multiple experiments

### ISSUE THREAD

**Issue by 1602353775** [embeddings-benchmark/mteb#3897]
### Description

In classification task evaluation, the `_undersample_data` method receives the `experiment_num` parameter but does not use it when initializing the random state. This causes all experiments (`n_experiments`) to use the same random seed, resulting in identical or highly similar sampling results, which prevents true Bootstrap sampling.

### Current Behavior

In the `_evaluate_subset` method of `mteb/abstasks/classification.py`, `_undersample_data` is called in a loop for multiple experiments:

for i in range(self.n_experiments):
    train_dataset, idxs = self._undersample_data(
        train_split,
        i,  # experiment_num is passed
        idxs,
    )
rng_state = np.random.RandomState(self.seed)  # experiment_num is not used
rng_state.shuffle(idxs)

### To reproduce

# Current code
rng_state = np.random.RandomState(self.seed)

# Suggested fix
rng_state = np.random.RandomState(self.seed + experiment_num)

### Are you interested to contribute a fix for this bug?

No
